ProcessLib/ThermoHydroMechanics/CreateThermoHydroMechanicsProcess.cpp

```diff
@@ -0,0 +1,297 @@
+        "porosity", parameters, 1);
+    DBUG("Use \'%s\' as porosity parameter.", porosity.name.c_str());
+
+    // Solid density
+    auto& solid_density = findParameter<double>(
+        config,
+        //! \ogs_file_param_special{prj__processes__process__THERMO_HYDRO_MECHANICS_solid_density}
+        "solid_density", parameters, 1);
+    DBUG("Use \'%s\' as solid density parameter.", solid_density.name.c_str());
+
+    // Fluid density
+    auto const& fluid_density = findParameter<double>(
+        config,
+        //! \ogs_file_param_special{prj__processes__process__THERMO_HYDRO_MECHANICS_fluid_density}
+        "fluid_density", parameters, 1);
+    DBUG("Use \'%s\' as fluid density parameter.", fluid_density.name.c_str());
+
+    // linear thermal expansion coefficient for solid
+    auto const& solid_linear_thermal_expansion_coefficient = findParameter<double>(
+        config,
+        //! \ogs_file_param_special{prj__processes__process__THERMO_HYDRO_MECHANICS_solid_linear_thermal_expansion_coefficient}
+        "solid_linear_thermal_expansion_coefficient", parameters, 1);
+    DBUG("Use \'%s\' as solid linear thermal expansion coefficient parameter.",
+         solid_linear_thermal_expansion_coefficient.name.c_str());
+
+    // volumetric thermal expansion coefficient for fluid
+    auto const& fluid_volumetric_thermal_expansion_coefficient = findParameter<double>(
+        config,
+        //! \ogs_file_param_special{prj__processes__process__THERMO_HYDRO_MECHANICS_fluid_volumetric_thermal_expansion_coefficient}
+        "fluid_volumetric_thermal_expansion_coefficient", parameters, 1);
+    DBUG("Use \'%s\' as fluid volumetric thermal expansion coefficient parameter.",
+         fluid_volumetric_thermal_expansion_coefficient.name.c_str());
+
+    // specific heat capacity for fluid
+    auto& fluid_heat_capacity = findParameter<double>(
+        config,
+        //! \ogs_file_param_special{prj__processes__process__THERMO_HYDRO_MECHANICS_fluid_heat_capacity}
+        "fluid_heat_capacity", parameters, 1);
+    DBUG("Use \'%s\' as fluid heat capacity parameter.",
+         fluid_heat_capacity.name.c_str());
+
+    // specific heat capacity for solid
+    auto& solid_heat_capacity = findParameter<double>(
+        config,
+        //! \ogs_file_param_special{prj__processes__process__THERMO_HYDRO_MECHANICS_solid_heat_capacity}
+        "solid_heat_capacity", parameters, 1);
+    DBUG("Use \'%s\' as solid heat capacity parameter.",
+         solid_heat_capacity.name.c_str());
+
+    // thermal conductivity for solid
+    auto& solid_thermal_conductivity = findParameter<double>(
+        config,
+        //! \ogs_file_param_special{prj__processes__process__THERMO_HYDRO_MECHANICS_solid_thermal_conductivity}
+        "solid_thermal_conductivity", parameters, 1);
+    DBUG("Use \'%s\' as solid thermal conductivity parameter.",
+         solid_thermal_conductivity.name.c_str());
+
```

Comment:
thermal conductivity and permeability are tensors ... either transform to tensor input or make note that process currently only supports isotropy.